### ISSUE TITLE
Wrong Bash heredoc syntax highlighting

### ISSUE THREAD

**Issue by xroberx** [helix-editor/helix#7050]
### Summary

Syntax highlighting for Bash heredoc is broken. 

Sample Helix screenshots (incorrect): 

![scr01_hx](https://github.com/helix-editor/helix/assets/1663186/091d0fb9-b769-40de-9e34-3587a7eb29b5)
![scr02_hx](https://github.com/helix-editor/helix/assets/1663186/eb3393b9-b808-4847-a42f-8f38abc1f2a9)

Sample Visual Studio Code screenshots (correct): 

![scr01_vscode](https://github.com/helix-editor/helix/assets/1663186/e136f87a-a5a9-4d0c-916a-b761d0c3bae0)
![scr02_vscode](https://github.com/helix-editor/helix/assets/1663186/9f18b523-7093-4a35-a08d-a75ad7f411bc)

### Reproduction Steps

Open any bash script with heredoc syntax

### Platform

Linux / Windows

### Terminal Emulator

xfce4-terminal / Windows Terminal

### Helix Version

helix 23.03

**Comment by David-Else** [comment]
That will down down to https://github.com/tree-sitter/tree-sitter-bash. It would be better to file an issue there.

**Comment by pascalkuthe** [comment]
I also suspect this is an upstream issue but could you also upload the file here to make it easy to reprduicr the issue?

**Comment by xroberx** [comment]
You are right: https://github.com/tree-sitter/tree-sitter-bash/issues/159#issuecomment-1482214316
